### PR TITLE
feat(REST): Search package by purl and version

### LINKED ISSUE #2445: REST API endpoint for search packages by version and purl.
### Description

Currently, packages are searchable via name and purl only through REST. Extend this to include version and purl.

### PR DESCRIPTION
**Pull request by afsahsyeda**
### Description

New endpoint called "Search Packages" has been created. It enables the user to search for packages through name, version, purl or packageManagers and also perform more specific searches by combining these parameters using the & operator.
Closes #2445 

image

### CODE REVIEW

**Comment by afsahsyeda** [comment]
@sameed20 please perform a test on this PR.

**Comment by sameed20** [comment]
Test Successful

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/packages/PackageController.java [hunk 1]
```diff
@@ -228,25 +232,31 @@ public ResponseEntity<?> getPackagesForUser(
         String queryString = request.getQueryString();
         Map<String, String> params = restControllerHelper.parseQueryString(queryString);
         boolean isSearchByName = name != null && !name.isEmpty();
-        boolean isSearchByType = CommonUtils.isNotNullEmptyOrWhitespace(packageManager);
+        boolean isSearchByPackageManager = CommonUtils.isNotNullEmptyOrWhitespace(packageManager);
+        boolean isSearchByVersion = version != null && !version.isEmpty();
```
Comment (smrutis1) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/packages/PackageController.java:236]: Cant we use the same `CommonsUtils.isNotNullEmptyWhitespace(_)` as in line number `235`

**Review by smrutis1 — COMMENTED** [review]
one minor change requested

**Comment by afsahsyeda** [comment]
@smrutis1 I have made the suggested changes, please check.

**Review by smrutis1 — APPROVED** [review]
Code looks good to me.

**Pull request merged**